src/main/java/frc/robot/Constants.java

```diff
@@ -104,6 +104,104 @@ public static final class ModuleConstants {
+            CHAIN_0(new Translation2d(0, 0), new Translation2d(0, 0), new Translation2d(0, 0)),
+            CHAIN_120(new Translation2d(0, 0), new Translation2d(0, 0), new Translation2d(0, 0)),
+            CHAIN_240(new Translation2d(0, 0), new Translation2d(0, 0), new Translation2d(0, 0)),
+            CHAIN_60(new Translation2d(0, 0), new Translation2d(0, 0), new Translation2d(0, 0)),
+            CHAIN_180(new Translation2d(0, 0), new Translation2d(0, 0), new Translation2d(0, 0)),
+            CHAIN_300(new Translation2d(0, 0), new Translation2d(0, 0), new Translation2d(0, 0));
+
+            public final Translation2d negEdge;
+            public final Translation2d posEdge;
+            public final Translation2d centerStage;
+
+            Chain(Translation2d negEdge, Translation2d posEdge, Translation2d centerStage) {
+                this.negEdge = negEdge;
+                this.posEdge = posEdge;
+                this.centerStage = centerStage;
+            }
+
+            //gets the robot translation, returns true if the robot's angle from
+            //the center of the stage is in the range of angles defined for this chain
+            public boolean inRange(Translation2d robotTranslation) {
+                //calculate the robot's angle from the center of the stage (relative to the x axis)
+                double robotStageAngle =
+                        Math.atan(robotTranslation.minus(centerStage).getX() / robotTranslation.minus(centerStage).getY());
+                //calculate the same thing for the neg and pos edges
+                double negAngle = Math.atan(negEdge.minus(centerStage).getX() / negEdge.minus(centerStage).getY());
+                double posAngle = Math.atan(posEdge.minus(centerStage).getX() / posEdge.minus(centerStage).getY());
+                //returns true if the robot angle is between the neg and pos angles
+                if (posAngle > negAngle) return robotStageAngle < posAngle && robotStageAngle > negAngle;
+                return (robotStageAngle > posAngle && robotStageAngle > negAngle)
+                        || (robotStageAngle < posAngle && robotStageAngle < negAngle);
+            }
+
+            //this function returns the chain that is both in
+            //the robot's color in returns true to the inRange function
+            public static Chain getBestChain(Translation2d robotTranslation) {
+                //recognise which chain we want to climb on
+                if (isBlueAlliance()) {
+                    if (Chain.CHAIN_0.inRange(robotTranslation)) return Chain.CHAIN_0;
+                    else if (Chain.CHAIN_120.inRange(robotTranslation)) return Chain.CHAIN_120;
+                    return Chain.CHAIN_240;
+                }
+
+                if (Chain.CHAIN_60.inRange(robotTranslation)) return Chain.CHAIN_60;
+                else if (Chain.CHAIN_180.inRange(robotTranslation)) return Chain.CHAIN_180;
+                return Chain.CHAIN_300;
+            }
```

Comment:
this wouldn't be for the robot to decide on it's own, the driver will choose the chain to climb on depending on other robots, and other factors,